File: kotlin/src/test/kotlin/com/goblindegook/fizzbuzz/FizzbuzzTest.kt
package com.goblindegook.fizzbuzz

import kotlin.test.Test
import kotlin.test.assertEquals

class FizzbuzzTest {
  @Test
  fun `multiples of 3 are fizz`() {
    assertEquals(listOf("fizz"), fizzbuzz(3, 3))
  }

  @Test
  fun `multiples of 5 are buzz`() {
    assertEquals(listOf("buzz"), fizzbuzz(5, 5))
  }

  @Test
  fun `multiples of 15 are fizzbuzz`() {
    assertEquals(listOf("fizzbuzz"), fizzbuzz(15, 15))
  }

  @Test
  fun `other numbers are unchanged`() {
    assertEquals(listOf("1"), fizzbuzz(1, 1))
  }

  @Test
  fun `processes ranges`() {
    assertEquals(listOf("fizz", "-2", "-1", "fizzbuzz", "1", "2", "fizz"), fizzbuzz(-3, 3))
  }

  @Test
  fun `processes descending ranges`() {
    assertEquals(listOf("1", "fizzbuzz", "-1"), fizzbuzz(1, -1))
  }
}

File: elixir/test/fizzbuzz_test.exs
defmodule FizzbuzzTest do
  use ExUnit.Case
  doctest Fizzbuzz

  test "converts multiples of 3 to fizz" do
    assert Fizzbuzz.fizzbuzz(3, 3) == ["fizz"]
  end

  test "converts multiples of 5 to buzz" do
    assert Fizzbuzz.fizzbuzz(5, 5) == ["buzz"]
  end

  test "converts multiples of 3 and 5 to fizzbuzz" do
    assert Fizzbuzz.fizzbuzz(15, 15) == ["fizzbuzz"]
  end

  test "other numbers are unchanged" do
    assert Fizzbuzz.fizzbuzz(1, 1) == [1]
  end

  test "converts ranges" do
    assert Fizzbuzz.fizzbuzz(0, 5) == ["fizzbuzz", 1, 2, "fizz", 4, "buzz"]
  end
end
